src/components/ToDoContainer/index.js

```diff
@@ -1,11 +1,77 @@
-import React, { Component } from "react";
+import React, {
+  Component
+} from "react";
+
+import NewTodoForm from './NewTodoForm';
+import TodoList from './TodoList';
 
 class ToDoContainer extends Component {
+  constructor() {
+    super();
+    this.state = {
+      newTodo: '',
+      todos: []
+    }
+
+    this.formSubmitted = this.formSubmitted.bind(this);
+    this.newTodoChanged = this.newTodoChanged.bind(this);
+    this.removeTodo = this.removeTodo.bind(this);
+    this.searchTodos = this.searchTodos.bind(this);
+  }
+
+  // Create new todo
+  newTodoChanged(event) {
+    this.setState({
+      newTodo: event.target.value
+    })
+
+  }
+
+  // Form submission event handler
+  formSubmitted(event) {
+    event.preventDefault();
+    // Add new todo to array
+    this.setState({
+      newTodo: '',
+      todos: [...this.state.todos, {
+        title: this.state.newTodo,
+        done: false
+      }]
+    })
+  }
+
+  // Remove clicked todolist
+  removeTodo(index) {
+    const todos = [...this.state.todos];
+    todos.splice(index, 1);
+    this.setState({
+      todos
+    })
+  }
+
+  // Search todolist by substring
+  searchTodos(event) {
```

Comment:
By the way you currently implement this searching functionality, there is no way of recovering the results that don't match the query once you exit from the query. This is because you set the state as part of your searching. Instead consider this option:

You can store a `filter` property in the state of the `ToDoContainer`, which is updated by the input in the `ToDoList` whenever it is changed. Then, instead of changing the `todos` in the state, you can just conditionally render only the todos in your list which have a match with `this.state.filter`